pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -253,6 +256,11 @@ protected void consumeLoop() {
 
         if (row != null) {
           row = _fieldExtractor.transform(row);
+
```

Comment:
We could. The metric is ambiguous in the sense that at that point, the row is technically consumed from Kafka though not indexed yet. In practice, both are equivalent since indexing takes so little time.